winrt.lua

```diff
@@ -50,14 +50,27 @@
 
 	p.override(p.vstudio.vc2010, "emitFiles", function(base, prj, group, tag, fileFunc, fileCfgFunc, checkFunc)
 		if isWinRT(prj.system) then
-			if tag == "ClCompile" then
-				fileCfgFunc = table.join(fileCfgFunc or {}, {
-					m.compileAsWinRT,
-				})
-			elseif tag == "None" then
-				fileCfgFunc = table.join(fileCfgFunc or {}, {
-					m.deploymentContent,
-				})
+			local oldFn = fileCfgFunc
+			fileCfgFunc = function (...)
+				local elements
+
+				if type(oldFn) == "function" then
```

Comment:
Yes, it might be nil, a function or a table. Sometimes, the function overhead can be avoided because the same table is returned regardless of the configuration. I could have done that here too since I threw away the function parameters. Ultimately, it would be better if it was possible to override the fileCfgFunc function directly, but that requires a bit of refactoring in premake-core.